### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by reeichert**
Added some documentation, still missing some classes and methods. @dreymonde please review and update if needed.

### CODE REVIEW

**Comment by dreymonde** [comment]
Thanks for your contribution! I appreciate your work, but I feel like some parts of this pull request are out of its scope — I’m mostly talking about changes in code style and formatting

File: Sources/AppFolder/BaseFolder.swift [hunk 1]
```diff
@@ -8,55 +8,95 @@
...
 
+/// Protocol that define default methods of a AppGroup.
 public protocol AppGroup {
-    
     static var groupIdentifier: String { get }
-    
 }
 
-public final class AppGroupContainer<Group : AppGroup> : BaseFolder {
+public final class AppGroupContainer<Group: AppGroup>: BaseFolder {
```
Comment (dreymonde) [inline Sources/AppFolder/BaseFolder.swift]: For example, changes like this: `<Group : AppGroup>` is not something that needs to be fixed, that's my code style (based on one of Swift **stdlib**), and I would like it to stay as it was 🙂

File: Sources/AppFolder/BaseFolder.swift [hunk 2]
```diff
@@ -8,55 +8,95 @@
...
         return url
     }()
     
 }
 
+/// Protocol that define default methods of a AppGroup.
 public protocol AppGroup {
-    
     static var groupIdentifier: String { get }
-
```
Comment (dreymonde) [inline Sources/AppFolder/BaseFolder.swift]: I’m also talking about this: it’s small and I don’t have very strong feelings about this, but still its unrelated to the scope of this pull request

**Review by dreymonde — COMMENTED** [review]
Once those style changes are reverted, I’m happy to merge this PR 😄
They are present in many places, but I’ll just outline a few of them here so that you can get the idea of what do I mean

File: Sources/AppFolder/BaseFolder.swift (hunk 1 above)
Comment (reeichert) [reply Sources/AppFolder/BaseFolder.swift]: Of course, I think I made those changes basically for habit, I will revert this

**Comment by reeichert** [comment]
@dreymonde please review.

**Comment by dreymonde** [comment]
Everything looks great now! Thanks for your contribution

**Pull request merged**